test_data/proxy/src/com/intellij/memory/agent/MemoryAgent.java

```diff
@@ -0,0 +1,47 @@
+package com.intellij.memory.agent;
+
+
+import com.intellij.memory.agent.proxy.IdeaNativeAgentProxy;
+import java.io.File;
+
+public class MemoryAgent {
+    private static MemoryAgent INSTANCE = null;
+    private final IdeaNativeAgentProxy proxy;
+
+    private MemoryAgent() {
+        proxy = new IdeaNativeAgentProxy();
+    }
+
+    public static synchronized MemoryAgent get() throws MemoryAgentException {
```

Comment:
It's possible to have lock on `Singleton.get()` quite easily in Java - see https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom